osf/management/commands/migrate_deleted_date.py

```diff
@@ -0,0 +1,96 @@
+    'addons_dropbox_usersettings',
+    'addons_dropbox_nodesettings',
+    'addons_figshare_nodesettings',
+    'addons_figshare_usersettings',
+    'addons_forward_nodesettings',
+    'addons_github_nodesettings',
+    'addons_github_usersettings',
+    'addons_gitlab_nodesettings',
+    'addons_gitlab_usersettings',
+    'addons_googledrive_nodesettings',
+    'addons_googledrive_usersettings',
+    'addons_mendeley_nodesettings',
+    'addons_mendeley_usersettings',
+    'addons_onedrive_nodesettings',
+    'addons_onedrive_usersettings',
+    'addons_osfstorage_nodesettings',
+    'addons_osfstorage_usersettings',
+    'addons_bitbucket_nodesettings',
+    'addons_bitbucket_usersettings',
+    'addons_owncloud_nodesettings',
+    'addons_box_nodesettings',
+    'addons_owncloud_usersettings',
+    'addons_box_usersettings',
+    'addons_dataverse_nodesettings',
+    'addons_dataverse_usersettings',
+    'addons_s3_nodesettings',
+    'addons_s3_usersettings',
+    'addons_twofactor_usersettings',
+    'addons_wiki_nodesettings',
+    'addons_zotero_nodesettings'
+]
+
+UPDATE_DELETED_WITH_MODIFIED = 'SET statement_timeout = 10000; UPDATE {} SET deleted=modified WHERE id IN (SELECT id FROM {} WHERE is_deleted AND deleted IS NULL LIMIT {}) RETURNING id;'
+
+
+@celery_app.task(name='management.commands.migrate_deleted_date')
+def populate_deleted(dry_run=False, page_size=1000):
+    with transaction.atomic():
+        for table in TABLES_TO_POPULATE_WITH_MODIFIED:
+            run_statements(UPDATE_DELETED_WITH_MODIFIED, page_size, table)
+        if dry_run:
+            raise RuntimeError('Dry Run -- Transaction rolled back')
+
+def run_statements(statement, page_size, table):
+    logger.info('Populating deleted column in table {}'.format(table))
+    with connection.cursor() as cursor:
+        cursor.execute(statement.format(table, table, page_size))
+        rows = cursor.fetchall()
+        if not rows:
+            logger.info('Sentry notification that {} is populated'.format(table))
+
+class Command(BaseCommand):
+    help = '''Populates new deleted field for various models. Ensure you have run migrations
+    before running this script.'''
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--dry_run',
+            type=bool,
+            default=False,
+            help='Run queries but do not write files',
+        )
+        parser.add_argument(
+            '--page_size',
+            type=int,
+            default=1000,
+            help='How many rows to process at a time',
+        )
+
+    def handle(self, *args, **options):
+        script_start_time = datetime.datetime.now()
+        logger.info('Script started time: {}'.format(script_start_time))
+        logger.debug(options)
+
+        dry_run = options['dry_run']
+        page_size = options['page_size']
+
+        if dry_run:
+            logger.info('DRY RUN')
+
+        populate_deleted(dry_run, page_size)
```

Comment:
The tricky thing with moving items like these into management commands is that this data really needs to get populated on every environment, and (unlike migrations) there's nothing enforcing that this happens.    

Double check with @mfraezz, but since we'll probably have to take downtime already to add these new columns to the addon tables, I would just move this management command into a migration where we take care of all of it at once.  It is going to be quick, confirmed on prod data.

Reply:
No strong opinions just from glancing over it. I'm generally of the (sometimes controversial) opinion that schema migrations should be migrations, and data migrations should be scripts or commands, excepting cases where the app cannot run without having the data populated. (Aside: in these cases, it's best to follow a "read one, write both," and fail over when a cron is done back-populating.) 

However, if it's fast and _guaranteed_ to not break on any environment, it doesn't make much difference from a deployment perspective.

Flagging it by env is another option. ¯\\\_(ツ)_/¯